scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx

```diff
@@ -43,9 +43,11 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
         const {suggestions, message} = this.props.warning;
         const {spellchecker} = this.props;
 
-        // If the message exists, use it as the button text
-        // instead of the suggestion
+        // If the message exists, and suggestion is whitespace suggestion
+        // use message as the button text instead of the suggestion
         const messageExists = Boolean(message);
+        const isWhitespaceSuggestion = suggestions.length === 1
+            && (suggestions[0].text === ' ' || suggestions[0].text === '');
```

Comment:
What if there are more than one suggestions and only one of them is whitespace? What if whitespace is multiple characters instead of one? While the second one is unlikely, it's not hard to handle so why not? Use `trim().length > 0` to check for whitespace.